common/lib/client/paginatedresource.ts

```diff
@@ -0,0 +1,195 @@
+import * as Utils from '../util/utils';
+import Logger from '../util/logger';
+import Resource from './resource';
+import ErrorInfo from '../types/errorinfo';
+import { PaginatedResultCallback } from '../../types/utils';
+
+// TODO: Replace this once rest.js is converted to TypeScript
+type Rest = any;
+
+export type BodyHandler = (body: unknown, headers: Record<string, string>, packed?: boolean) => any;
+
+function getRelParams(linkUrl: string) {
+	const urlMatch = linkUrl.match(/^\.\/(\w+)\?(.*)$/);
+	return urlMatch && Utils.parseQueryString(urlMatch[2]);
```

Comment:
Should you be also checking `urlMatch.size > 2` ?

Reply:
2e1fd3b1

Reply:
Resolving this conversation on @ikbalkaya's behalf.